### PR TITLE
fix: replace NODE_ENV for umd/iife (fix #274)

### LINKED ISSUE #274: Replacing `process.env.NODE_ENV`
I am aware of #19 however this functionality appears to have been removed.

I have tried implementing this manually using [`babel-plugin-transform-define`](https://github.com/FormidableLabs/babel-plugin-transform-define) by adding the following to my `babel.config.js` plugins config:

```js
["babel-plugin-transform-define", {
      "process.env.NODE_ENV": "production",
}]
```

However this seems to be having no effect on the compiled output, I could do this via [`@rollup/plugin-replace`](https://github.com/rollup/plugins/tree/master/packages/replace) as demonstrated in this [comment](https://github.com/rollup/rollup/issues/487#issuecomment-177596512) but doing it via `babel` will convert things like `process.env.NODE_ENV === "production"` to `true` in the compiled output which I prefer.

Cheers.

**UPDATE**

Using:

```js
  plugins: {
    vue: true,
    json: true,
    image: true,
    replace: {
      'process.env.NODE_ENV': JSON.stringify( 'production' )
    }
  },
```

Still does not effect compiled output.

### PR DESCRIPTION
**Pull request by fi3ework**
- [x] test passed
- [x] snapshot added

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -282,6 +282,11 @@ export class Bundler {
 
     const env = Object.assign({}, config.env)
 
+    // drop precess.env.NODE_ENV from umd/iife
+    if (['umd', 'umd-min', 'iife', 'iife-min'].includes(format)) {
+      env.NODE_ENV = minify ? 'production' : 'development'
+    }
```
Comment (egoist) [inline src/index.ts:288]: we can skip this if `env.NODE_ENV` is already set,

Comment (fi3ework) [reply src/index.ts:291]: done

File: src/index.ts [hunk 2]
```diff
@@ -282,6 +282,14 @@ export class Bundler {
 
     const env = Object.assign({}, config.env)
 
+    // drop precess.env.NODE_ENV from umd/iife
```
Comment (chocolateboy) [inline src/index.ts:285]: typo: precess -> process

**Pull request merged**